src/backend/utils/adt/like_match.c

```diff
@@ -350,15 +350,26 @@ do_like_escape(text *pat, text *esc)
 
 	if (elen == 0)
 	{
-		/*
-		 * No escape character is wanted.  Double any backslashes in the
-		 * pattern to make them act like ordinary characters.
-		 */
-		while (plen > 0)
-		{
-			if (*p == '\\')
-				*r++ = '\\';
-			CopyAdvChar(r, p, plen);
+		if (sql_dialect == SQL_DIALECT_TSQL){
```

Comment:
if the escape string is an empty string, the elen will be 0. if the escape string is null, it will not go through the do_like_escape function. So I think you only need this sql_dialect check to solve this issue.

Reply:
Updated